qbraid/runtime/rigetti/device.py

```
@@ -89,20 +98,123 @@ def __str__(self) -> str:
         except ListQuantumProcessorsError as e:
             raise RigettiDeviceError(  # pylint: disable=bad-exception-cause
                 "Failed to retrieve quantum processor list from Rigetti QCS."
             ) from e
+
+        if self.id not in quantum_processor_ids:
+            return DeviceStatus.OFFLINE
+
+        try:
+            if availability.is_in_maintenance(self._fetch_maintenance_ical()):
+                return DeviceStatus.UNAVAILABLE
+        except Exception as e:  # pylint: disable=broad-exception-caught
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)